herbert_core/utilities/classes/@object_lookup/object_lookup.m

```diff
@@ -149,484 +182,374 @@
             
-            if numel(val)==numel(obj.object_store_) || isscalar(val)
-                if numel(obj.object_store_)>0
-                    if numel(val)==numel(obj.object_store_)
-                        obj.object_store_ = val(:);
-                    else
-                        obj.object_store_ = repmat(val(:),size(obj.object_store_));
-                    end
-                else
-                    % Force default null object_store if currently unassigned
-                    null = object_lookup;
-                    obj.object_store = null.object_store_;
-                end
-            else
+            % The object store is turned into a column vector. An empty object
+            % array is permitted.
+            % Duplicate elements are removed and the corresponding index arrays
+            % in property indx are updated in the call to check_combo_arg
+            
+            if ~isobject(val)
+                error('HERBERT:object_lookup:invalid_argument', ...
+                    'Property ''object_store'' must be a Matlab object or array of objects')
+            end
+
+            obj.object_store_ = val(:);     % ensure column vector
+            
+            % Check interdependencies
+            if obj.do_check_combo_arg_
+                obj = obj.check_combo_arg();
+            end
+        end
+        
+        
+        function obj = set.indx (obj, val)
+            % Set index arrays into the object_store
+            %
+            %   >> obj.indx = val
+            %
+            %   val     Index array(s) into the unique object store of the
+            %           stored object array(s):
+            %           - Numeric array if there is a single stored object array
+            %           - Cell array of index arrays, where val{i} is the vector
+            %             of indices into the store for the ith object array
+            %
+            % NOTE:
+            % - val will be stored as a column callarray of column vectors,
+            %   which is how it will be returned by the corresponding getter:
+            %       >> val = obj.indx
+            
+            % It is valid for an index array to be empty: this corresponds to an
+            % empty stored object array
+            
+            % The following permits empty index arrays to pass the test, thereby
+            % permitting the default object (i.e. constructor called with no
+            % arguments) to pass.
+            if ~iscell(val)
+                val = {val};    % turn into a cell array with one element
+            end
+            positive_int_array = @(x)(all(x>=1,'all') && all(rem(x,1)==0, 'all'));
+            if ~all (cellfun (@isnumeric, val), 'all') ||...
+                    ~all (cellfun (@(x)positive_int_array(x), val), 'all')
+                error('HERBERT:object_lookup:invalid_argument', ...
+                    'Property ''indx'' must be a cell array of integer arrays')
+            end
```

Comment:
```suggestion
            positive_int_array = @(x)(isnumeric(x) && all(x>=1,'all') && all(rem(x,1)==0, 'all'));
            if ~all (cellfun (@(x)positive_int_array(x), val), 'all')
                error('HERBERT:object_lookup:invalid_argument', ...
                    'Property ''indx'' must be a cell array of integer arrays')
            end
```

Reply:
The suggested code does not have the desired functionality. The original code first checks that all elements of the cellarray are numeric; only then does it test that they are integer. In the suggested code  the comparison all(x>=1,'all')  would be attempted regardless of the classes of the elements of the cellarray, which depending on the class could throw an error (e.g. an element itself being a cell array) or unexpected behaviour (e.g. character array).